internal/service/common/certificatesigningrequest.go

```diff
@@ -0,0 +1,63 @@
+package common
+
+import (
+	"context"
+	"errors"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/api/server"
+	"github.com/flightctl/flightctl/internal/flterrors"
+	"github.com/flightctl/flightctl/internal/store"
+)
+
+func ValidateAndCompleteCertificateSigningRequest(certificateSigningRequest *v1alpha1.CertificateSigningRequest) error {
+	if certificateSigningRequest.Status == nil {
+		certificateSigningRequest.Status = &v1alpha1.CertificateSigningRequestStatus{
+			Certificate: nil,
+			Conditions:  []v1alpha1.Condition{},
+		}
+	}
+	return nil
+}
+
+func CreateCertificateSigningRequest(ctx context.Context, st store.Store, request server.CreateCertificateSigningRequestRequestObject) (server.CreateCertificateSigningRequestResponseObject, error) {
+	orgId := store.NullOrgId
+
+	// don't set fields that are managed by the service
+	request.Body.Status = nil
+	NilOutManagedObjectMetaProperties(&request.Body.Metadata)
+
+	if errs := request.Body.Validate(); len(errs) > 0 {
+		return server.CreateCertificateSigningRequest400JSONResponse{Message: errors.Join(errs...).Error()}, nil
+	}
+
+	// note: this is not checking whether the signing request already exists; it will overwrite an existing request
+
+	if err := ValidateAndCompleteCertificateSigningRequest(request.Body); err != nil {
+		return nil, err
+	}
+
+	result, err := st.CertificateSigningRequest().Create(ctx, orgId, request.Body)
+	switch err {
+	case nil:
+		return server.CreateCertificateSigningRequest201JSONResponse(*result), nil
+	case flterrors.ErrResourceIsNil, flterrors.ErrIllegalResourceVersionFormat:
+		return server.CreateCertificateSigningRequest400JSONResponse{Message: err.Error()}, nil
+	default:
+		return nil, err
+	}
+}
+
+func ReadCertificateSigningRequest(ctx context.Context, st store.Store, request server.ReadCertificateSigningRequestRequestObject) (server.ReadCertificateSigningRequestResponseObject, error) {
```

Comment:
Should this be called from internal/service/certificatesigningrequest.go?

Reply:
I took this from enrollmentrequest, but yep that makes sense.